### PR TITLE
Run Tests in GitHub CI using actions instead of using play instance

### PR DESCRIPTION
**Pull request by prakashsvmx**
Run Tests in GitHub CI using actions instead of play.min.io

How to test:
Linux:
I have used https://github.com/nektos/act to run the actions locally and verify the status. 

Windows:
Passing as per the actions

Note:  Modified a functional test to remove windows specific special characters in object name.

### CODE REVIEW

**Review by ebozduman — CHANGES_REQUESTED** [review]

File: src/test/functional/functional-tests.js [hunk 1]
```diff
@@ -2520,8 +2520,9 @@ describe('functional tests', function() {
     after((done) => client.removeBucket(bucketNameForSpCharObjects, done))
 
     // Reference:: https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html
-    const objectNameSpecialChars="äöüex ®©µÄÆÐÕæŒƕƩǅ 01000000 0x40 \u0040 amȡȹɆple&0a!-_.*'()&$@=;:+,?<>.pdf"
-
+    // Host OS compatible File name characters/ file names.
+    // const objectNameSpecialChars="äöüex ®©µÄÆÐÕæŒƕƩǅ 01000000 0x40 \u0040 amȡȹɆple&0a!-_.*'()&$@=;:+,?<>.pdf" // this is valid on linux.
+    const objectNameSpecialChars="äöüex ®©µÄÆÐÕæŒƕƩǅ 01000000 0x40 u0040 amȡȹɆple&0a!-_.'()&$@=;+,.pdf"
```
[the remark this answers is not in the document]
Comment (prakashsvmx) [reply src/test/functional/functional-tests.js:2525]: Thank you. Yes. I will check the os and use the object name accordingly.

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
ping @kanagarajkm @ebozduman

**Comment by kanagarajkm** [comment]
@prakashsvmx Windows test is failing 

```
  1) functional tests
       Special Characters test on a prefix and an object
         putObject(bucketName, objectName, stream)_bucketName:minio-js-test-obj-sppre22d99c8b-c3b2-4fe9-b5dc-4e323810d03d, _objectName:SpecialMenùäöüexPrefix/äöüex ®©µÄÆÐÕæŒƕƩǅ 01000000 0x40 @ amȡȹɆple&0a!-_.*'()&$@=;:+,?<>.pdf, stream:100Kib:
     S3Error: Object name contains unsupported characters.
```

**Comment by harshavardhana** [comment]
> putObject(bucketName, objectName, stream)_bucketName:minio-js-test-obj-sppre22d99c8b-c3b2-4fe9-b5dc-4e323810d03d, _objectName:SpecialMenùäöüexPrefix/äöüex ®©µÄÆÐÕæŒƕƩǅ 01000000 0x40 @ amȡȹɆple&0a!-_.*'()&$@=;:+,?<>.pdf, stream:100Kib:

Windows won't support these characters @prakashsvmx you would have to skip this test on windows

**Comment by prakashsvmx** [comment]
@harshavardhana  @kanagarajkm  Thank you for the review.

`isWindowsPlatform` check  is introduced in this PR. 
 interesting finding is that old CI jobs for windows never failed for this test case.

File: .github/workflows/nodejs-windows.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          ACCESS_KEY: minio
+          SECRET_KEY: minio123
+          ENABLE_HTTPS: 0
+        run: |
+          Invoke-WebRequest -Uri https://dl.minio.io/server/minio/release/windows-amd64/minio.exe -OutFile $HOME/minio.exe
+          Start-Process -NoNewWindow -FilePath "$HOME/minio.exe" -ArgumentList "-C", "$env:temp/minio-config", "server", "$env:temp/data{1...4}"
+          npm install
+          npm run lint
+          npm run browserify
+          npm run test
```
Comment (harshavardhana) [inline .github/workflows/nodejs-windows.yml:40]:
```suggestion
          npm run test

```

File: .github/workflows/nodejs.yml [hunk 3]
```diff
@@ -1,33 +1,43 @@
...
+          SECRET_KEY: minio123
+          ENABLE_HTTPS: 0
+        run: |
+          wget --quiet -O /tmp/minio https://dl.min.io/server/minio/release/linux-amd64/minio
+          chmod +x /tmp/minio
+          /tmp/minio -C /tmp/minio-config server /tmp/data{1...4} &
+          npm install
+          npm run lint
+          npm run browserify
+          npm run test
```
Comment (harshavardhana) [inline .github/workflows/nodejs.yml:43]:
```suggestion
          npm run test

```

**Review by harshavardhana — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Review by ebozduman — APPROVED** [review]
LGTM

**Pull request merged**